opentelemetry-api/src/opentelemetry/trace/__init__.py

```diff
@@ -509,43 +540,45 @@ def use_span(
 # the following type definition should be replaced with
 # from opentelemetry.util.loader import ImplementationFactory
 ImplementationFactory = typing.Callable[
-    [typing.Type[Tracer]], typing.Optional[Tracer]
+    [typing.Type[TracerSource]], typing.Optional[TracerSource]
 ]
 
-_TRACER = None  # type: typing.Optional[Tracer]
-_TRACER_FACTORY = None  # type: typing.Optional[ImplementationFactory]
+_TRACER_SOURCE = None  # type: typing.Optional[TracerSource]
+_TRACER_SOURCE_FACTORY = None  # type: typing.Optional[ImplementationFactory]
```

Comment:
I just did a straight port from the old code. Because with the same argument you could have put the tracer instance here. I guess the reason I implemented it that way is that the loader supports overriding the `set_preferred_*_implementation` call with an environment variable and then your constructed tracer source would be "thrown away", potentially doing an expensive initialization for nothing.